src/paren.py

```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+
+from linked_list import LinkedList
```

Comment:
In your implementation below, does using a Linked list give you any benefit that just using the `string_to_check` and iterating over it would not give you all by itself?  Don't use a data structure just for fun.  Use one because it allows you to do something you wouldn't be able to do otherwise.